### PR TITLE
spectator entity location based on eye level (WIP)

### PR DESCRIPTION
**Pull request by bradcarnage**
(static "cat" string needs to be swapped out for either via config load or some other method)

### CODE REVIEW

File: src/main/java/com/zenith/network/server/handler/spectator/incoming/movement/PlayerPositionRotationSpectatorHandler.java [hunk 1]
```diff
@@ -29,31 +31,47 @@ public static void updateSpectatorPosition(final ServerConnection selfSession) {
         if (selfSession.isPlayerCam()) {
             return;
         }
+        double playerEyeHeight = 1.62;
+        EntityStats entityStats = new EntityStats();
```
Comment (rfresh2) [inline src/main/java/com/zenith/network/server/handler/spectator/incoming/movement/PlayerPositionRotationSpectatorHandler.java]: Integrate any additional metadata about spectator entity types in the SpectatorEntity class hierarchy under here: https://github.com/rfresh2/ZenithProxy/tree/1.20/src/main/java/com/zenith/feature/spectator

The current spectator entity type can be accessed here on the ServerConnection: https://github.com/rfresh2/ZenithProxy/blob/c04b583187cbb0dc5d745fd4bb5ec562f804cae6/src/main/java/com/zenith/network/server/ServerConnection.java#L78

so you shouldn't need any `EntityData` class initializations or hardcoded map lookups in the `updateSpectatorPosition` function.

I'm imagining we add a set of methods on SpectatorEntity to get whatever data is needed about the current type.

File: src/main/java/com/zenith/util/math/MathHelper.java [hunk 2]
```diff
@@ -44,4 +44,20 @@ public static float wrapDegrees(float degrees) {
...
+        double xOffset = -Math.sin(yawRadians) * Math.cos(pitchRadians) * distance;
+        double yOffset = -Math.sin(pitchRadians) * distance;
+        double zOffset = Math.cos(yawRadians) * Math.cos(pitchRadians) * distance;
+
+        double newX = x + xOffset;
+        double newY = y + yOffset;
+        double newZ = z + zOffset;
+
+        return new double[]{newX,newY,newZ};
+    }
```
Comment (rfresh2) [inline src/main/java/com/zenith/util/math/MathHelper.java]: I'd prefer inlining this to where its needed so we avoid heap allocations on the hotpath of packet processing. 

I don't think this func is being reused or shared anywhere else

**Review by rfresh2 — CHANGES_REQUESTED** [review]

File: src/main/java/com/zenith/util/math/MathHelper.java (hunk 2 above)
Comment (bradcarnage) [reply src/main/java/com/zenith/util/math/MathHelper.java]: change was made

File: src/main/java/com/zenith/network/server/handler/spectator/incoming/movement/PlayerPositionRotationSpectatorHandler.java (hunk 1 above)
Comment (bradcarnage) [reply src/main/java/com/zenith/network/server/handler/spectator/incoming/movement/PlayerPositionRotationSpectatorHandler.java]: change was made

**Review by rfresh2 — APPROVED** [review]

**Comment by rfresh2** [comment]
Thanks!

**Pull request merged**